plugin/manager/manager.go

```diff
@@ -0,0 +1,240 @@
+package manager
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/fs"
+	"os"
+	"os/exec"
+	"path"
+	"path/filepath"
+	"runtime"
+
+	"github.com/notaryproject/notation-go/plugin"
+)
+
+const (
+	// NamePrefix is the prefix required on all plugin binary names.
+	NamePrefix = "notation-"
+)
+
+// Plugin represents a potential plugin with all it's metadata.
+type Plugin struct {
+	plugin.Metadata
+
+	Path string `json:",omitempty"`
+
+	// Err is non-nil if the plugin failed one of the candidate tests.
+	Err error `json:",omitempty"`
+}
+
+// ErrNotFound is returned by Manager.Get and Manager.Run when the plugin is not found.
+var ErrNotFound = errors.New("plugin not found")
+
+// ErrNotCompliant is returned by Manager.Run when the plugin is found but not compliant.
+var ErrNotCompliant = errors.New("plugin not compliant")
+
+// ErrNotCapable is returned by Manager.Run when the plugin is found and compliant, but is missing a necessary capability.
+var ErrNotCapable = errors.New("plugin not capable")
+
+// commander is defined for mocking purposes.
+type commander interface {
+	Output(context.Context, string, ...string) ([]byte, error)
+}
+
+// execCommander implements the commander interface
+// using exec.Command().
+type execCommander struct{}
+
+func (c execCommander) Output(ctx context.Context, name string, args ...string) ([]byte, error) {
+	return exec.CommandContext(ctx, name, args...).Output()
+}
+
+// rootedFS is io.FS implementation used in NewManager.
+// root is the root of the file system tree passed to os.DirFS.
+type rootedFS struct {
+	fs.FS
+	root string
+}
+
+// Manager manages plugins installed on the system.
+type Manager struct {
+	fsys  fs.FS
+	cmder commander
+}
+
+// NewManager returns a new manager.
+func NewManager() *Manager {
```

Comment:
I would rather keep `NewManager` without parameters. Once we move the config management from `notation` to `notation-go` we will be able to root the manager to whatever path reports `config`, aka default plugins dir path. I expect this to be good enough for 95% of downstream users. If someone wants to use another config directory, we can provide `NewManagerWithPath(pluginsDir)` later on.

On the other hand, I can change the implementation now to use `os.UserConfigDir` instead of `os.UserHomeDir` so we support `XDG_CONFIG_HOME`.